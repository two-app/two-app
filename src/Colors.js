// @flow

const Colors = {
    SALMON: "#EE7674",
    DARK_SALMON: "#ed615f",
    LIGHT: "rgba(0, 0, 0, 0.1)",
    FADED: "rgba(0, 0, 0, 0.2)",
    DARK: "rgba(0, 0, 0, 0.8)",
    VALID_GREEN: "#85CC73"
};

export default Colors;