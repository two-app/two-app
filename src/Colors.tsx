const Colors = {
  SALMON: '#eb5754',
  DARK_SALMON: '#ed615f',
  LIGHT: 'rgba(0, 0, 0, 0.1)',
  FADED: 'rgba(0, 0, 0, 0.2)',
  REGULAR: 'rgb(100, 100, 100)',
  DARK: 'rgb(37, 37, 37)',
  VERY_DARK: 'rgb(21, 21, 21)',
  VALID_GREEN: '#3AB795',
  VALID_GREEN_DARK: '#36a88a',
};

export default Colors;
