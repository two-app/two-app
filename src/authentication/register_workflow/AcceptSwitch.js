import {StyleSheet, Switch, Text, View} from "react-native";
import PropTypes from "prop-types";
import Colors from "../../Colors";
import React, {useState} from "react";

const AcceptBox = ({children, onEmit}) => {
    const [accepted, setAccepted] = useState(false);
    return (
        <View style={styles.container}>
            <Text style={[styles.condition, accepted ? styles.acceptedCondition : undefined]}>{children}</Text>
            <View style={styles.switchContainer}>
                <Switch style={styles.switch}
                        value={accepted}
                        onValueChange={setAccepted}
                />
            </View>
        </View>
    );
};

AcceptBox.propTypes = {
    children: PropTypes.any,
    onEmit: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        flex: 1,
        justifyContent: "space-between",
        alignItems: "center",
        backgroundColor: Colors.LIGHT,
        borderRadius: 10,
        padding: 20,
        marginTop: 20
    },
    condition: {
        color: Colors.DARK,
        width: "80%"
    },
    acceptedCondition: {
        color: Colors.FADED
    },
    switchContainer: {
        width: "20%",
        flex: 1,
        alignItems: "center",
        justifyContent: "center"
    },
    switch: {
        transform: [{scaleX: .8}, {scaleY: .8}]
    }
});

export default AcceptBox;