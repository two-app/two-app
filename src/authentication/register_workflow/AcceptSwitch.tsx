import {StyleSheet, Switch, Text, View} from 'react-native';
import PropTypes from 'prop-types';
import React, {useState} from 'react';

import Colors from '../../Colors';

type AcceptBoxProps = {
  children?: any;
  onEmit: any;
  required?: boolean;
  accessibilityHint: string;
};

const AcceptBox = ({
  children,
  onEmit,
  required,
  accessibilityHint,
}: AcceptBoxProps) => {
  const [accepted, setAccepted] = useState(false);

  return (
    <View
      style={[
        styles.container,
        required ? styles.required : undefined,
        accepted ? styles.accepted : undefined,
      ]}
      data-testid="container">
      <Text
        style={[styles.condition, accepted ? styles.acceptedText : undefined]}>
        {children}
      </Text>
      <View style={styles.switchContainer}>
        <Switch
          style={styles.switch}
          value={accepted}
          onValueChange={v => {
            setAccepted(v);
            onEmit(v);
          }}
          accessibilityHint={accessibilityHint}
        />
      </View>
    </View>
  );
};

AcceptBox.propTypes = {
  onEmit: PropTypes.func.isRequired,
  isRequired: PropTypes.bool,
};

AcceptBox.defaultProps = {
  isRequired: false,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    borderRadius: 10,
    borderColor: Colors.LIGHT,
    borderWidth: 1,
    padding: 20,
    marginTop: 20,
  },
  required: {
    borderColor: Colors.DARK,
  },
  accepted: {
    borderColor: Colors.VALID_GREEN_DARK,
    backgroundColor: Colors.VALID_GREEN,
  },
  acceptedText: {
    color: 'white',
  },
  condition: {
    color: Colors.DARK,
    width: '80%',
  },
  switchContainer: {
    width: '20%',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  switch: {
    transform: [{scaleX: 0.8}, {scaleY: 0.8}],
  },
});

export default AcceptBox;
