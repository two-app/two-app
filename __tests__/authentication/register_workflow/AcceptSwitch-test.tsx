import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import type {ShallowWrapper} from 'enzyme';
import {shallow} from 'enzyme';

import AcceptSwitch from '../../../src/authentication/register_workflow/AcceptSwitch';
import Colors from '../../../src/Colors';

let tb: AcceptSwitchTestBed;

beforeEach(() => (tb = new AcceptSwitchTestBed()));

test('should maintain snapshot', () =>
  expect(
    renderer.create(
      <AcceptSwitch accessibilityHint="test hint" onEmit={tb.onEmitFn}>
        Some Condition
      </AcceptSwitch>,
    ),
  ).toMatchSnapshot());

test('should display the condition', () =>
  expect(tb.wrapper.find('Text').render().text()).toEqual('Some Condition'));

test('should emit when the value changes', () => {
  tb.setValueChanged();
  expect(tb.onEmitFn).toHaveBeenCalledWith(true);
});

describe('Switch', () => {
  test('should be false by default', () =>
    expect(tb.getSwitch().prop('value')).toBe(false));

  test('should change to true when clicked', () => {
    tb.setValueChanged();
    expect(tb.wrapper.find('Switch').prop('value')).toBe(true);
  });

  test('should turn the container border green', () => {
    tb.setValueChanged();

    expect(
      tb.doesContainerStyleContain('borderColor', Colors.VALID_GREEN_DARK),
    ).toBe(true);
  });

  test('should set the background to green', () => {
    tb.setValueChanged();
    expect(
      tb.doesContainerStyleContain('backgroundColor', Colors.VALID_GREEN),
    ).toBe(true);
  });
});

describe('Required', () => {
  test('should not be required by default, and have no border', () => {
    expect(tb.doesContainerStyleContain('borderColor', Colors.DARK)).toBe(
      false,
    );
  });

  test('should have border if required', () => {
    tb.wrapper.setProps({...tb.wrapper.props(), required: true});

    expect(tb.doesContainerStyleContain('borderColor', Colors.DARK)).toBe(true);
  });
});

class AcceptSwitchTestBed {
  onEmitFn = jest.fn();
  wrapper = shallow(
    <AcceptSwitch accessibilityHint="test hint" onEmit={this.onEmitFn}>
      Some Condition
    </AcceptSwitch>,
  );

  getSwitch = (): ShallowWrapper => this.wrapper.find('Switch');

  setValueChanged = () =>
    this.wrapper.find('Switch').prop<(v: boolean) => void>('onValueChange')(
      true,
    );

  doesContainerStyleContain = (key: string, value: any): boolean =>
    this.wrapper
      .find("View[data-testid='container']")
      .prop<[any]>('style')
      .filter(f => f !== undefined)
      .some(f => f[key] === value);
}
