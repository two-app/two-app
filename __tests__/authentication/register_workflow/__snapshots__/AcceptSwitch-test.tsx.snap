// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should maintain snapshot 1`] = `
<View
  data-testid="container"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "rgba(0, 0, 0, 0.1)",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
        "padding": 20,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgb(37, 37, 37)",
        "width": "80%",
      }
    }
  >
    Some Condition
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 0.8,
              },
              Object {
                "scaleY": 0.8,
              },
            ],
          },
        ]
      }
      value={false}
    />
  </View>
</View>
`;
